packages/back-end/src/controllers/metrics.ts

```diff
@@ -93,6 +91,11 @@ export async function deleteMetric(
     return;
   }
 
+  req.checkPermissions(
+    "createAnalyses",
+    metric?.projects?.length ? metric.projects : ""
```

Comment:
Is using an empty string here vs array intentional?

Reply:
The difference between passing in an empty string and an empty array only matters (currently) if the permission that's being checked is of the READ_ONLY_TYPE (e.g. runQueries, readData, or viewEvents).

In those cases, we're checking if the `checkProjects` that's defined in `userHasPermission` is equal to `[undefined]` - and if so, we're adding all of the projects the user has project-specific permissions for before checking `hasPermission`.

Looking at it this morning, the fix I have for this PR resolves the issue, however, I think refactoring `userHasPermission` to make it less opinionated on `""` vs `[]` might be the better route.

Changing the logic in `userHasPermission` from

```ts
if (
      checkProjects.length === 1 &&
      !checkProjects[0] === undefined &&
      Object.keys(userPermissions.projects).length
    ) 
```
to
```ts
if (
      checkProjects.length === 1 &&
      !checkProjects[0] &&
      Object.keys(userPermissions.projects).length
    ) 
```

This update makes it so it doesn't matter if the first index is an empty string or undefined, 

Alternatively, we could update the signature of `userHasPermission` so projects can only be undefined or a `string[]`.

In that case, for resources that can be in multiple projects (metrics, datasources, etc), we don't need the ternary. The only resources that would need to use the ternary are features, experiments, or ideas. In that case, we'd have to check for the existence of a project, and then either pass in an array with the project, or undefined.

This would make it so the engineer doesn't have to concern themselves with what they pass it. (Updating the logic so it just checks for `!checkProjects[0]` also removes that need, but an engineer still might wonder if they should pass in `""` or `[]`, even though it won't matter.

Digesting these options now...

Reply:
After consideration - for this PR, I think updating the check to simply check for `!checkProjects[0]` reduces the fragility.

I'd like to explore the larger fix that improves the developer experience, but would prefer to do so while there isn't a bug to ensure we test thoroughly.

Reply:
I think that change ("updating the check to simply check for `!checkProjects[0]`") makes sense to just do in this PR, with perhaps a bit more testing.

In the short-run, unless it breaks a bunch of permissions checks by making them more permissive when engineers built something and thought `""` would block permission, I would say we make it to reduce the possibility of these kinds of bugs. In any case, seeing as this distinction is only for `READ_ONLY_TYPE` permissions, then the potential blast radius of *incorrectly* making `""` pass more often seems small.